rcl/include/rcl/subscription.h

```diff
@@ -53,6 +53,15 @@ typedef struct rcl_subscription_options_s
   rmw_subscription_options_t rmw_subscription_options;
 } rcl_subscription_options_t;
 
+typedef struct rcl_subscription_content_filter_options_s
+{
+  /// Custom allocator for the options, used for incidental allocations.
+  /** For default behavior (malloc/free), see: rcl_get_default_allocator() */
+  rcl_allocator_t allocator;
```

Comment:
No, because the following two functions https://github.com/ros2/rcl/blob/cd8e85931a92f3bdb25fb879c6df43e03cfbba75/rcl/include/rcl/subscription.h#L395-L397 and  https://github.com/ros2/rcl/blob/cd8e85931a92f3bdb25fb879c6df43e03cfbba75/rcl/include/rcl/subscription.h#L426-L428 use this structure directly.

Reply:
But you're still not using this allocator there

https://github.com/ros2/rcl/blob/cd8e85931a92f3bdb25fb879c6df43e03cfbba75/rcl/src/rcl/subscription.c#L396

Reply:
Yeah, this line could be removed

Reply:
> Yeah, this line could be removed

Could you do that?
It seems that it's currently unused, and if we delete it we can make sure we're actually always using the same allocator (the one provided by `rcl_subscription_options_t`).

Reply:
What I said is to remove the line you mentioned at https://github.com/ros2/rcl/pull/894#discussion_r836522760, not to remove the `rcl_allocator_t allocator;` in the `rcl_subscription_content_filter_options_t`.

Reply:
Why not?
It's not being used anywhere (and if it is, use the one in `rcl_subscription_options_t` instead)

Reply:
Updated at 05811f0e5d1cbc72cca2abb212b3f06fb7e57f71